web/js/containers/sidebar/smart-handoff.js

```diff
@@ -0,0 +1,470 @@
+        topRight: util.formatCoordinate([geolonlat2[0], geolonlat2[1]]),
+      };
+
+      this.setState({
+        boundaries: newBoundaries,
+        coordinates,
+        currentExtent,
+      }, () => {
+        if (selectedLayer && currentExtent) {
+          this.debouncedUpdateExtent();
+        }
+      });
+    }
+  }
+
+  /**
+   * Fires when user selected a different layer
+   * @param {*} layer - the specified layer designated by the user
+   * @param {*} currentExtent - the current boundaries of the bounding box
+   */
+  onLayerChange(layer, currentExtent) {
+    this.setState({ selectedLayer: layer }, () => this.updateGranuleCount(currentExtent));
+  }
+
+  /**
+   * Asynchronous call to fetch granule data for the specified selected layer. Contains
+   * conditional logic to determine which counts are provided back to the user; total
+   * count vs selected count (that is if the bounding box has been enabled by the user)
+   * @param {*} currentExtent
+   */
+  async updateGranuleCount(currentExtent) {
+    const { selectedDate } = this.props;
+    const { selectedLayer, showBoundingBox } = this.state;
+
+    // Places the compoent state in a loading state; triggers {...} animation.
+    this.setState({ isSearchingForGranules: true });
+
+    const startDate = `${moment.utc(selectedDate).format('YYYY-MM-DD')}T00:00:00.000Z`;
+    const endDate = `${moment.utc(selectedDate).format('YYYY-MM-DD')}T23:59:59.999Z`;
+    const dateRange = `${startDate},${endDate}`;
+
+    let totalGranules = 0;
+    let selectedGranules = 0;
+
+    let urlTotalGranules = 'https://cmr.earthdata.nasa.gov/search/granules.json?'
+                + `temporal=${dateRange}&`
+                + `collection_concept_id=${selectedLayer.conceptId}&`
+                + 'include_facets=v2&'
+                + 'page_size=0';
+
+    if (selectedLayer.daynight) urlTotalGranules += `&day_night_flag=${selectedLayer.daynight}`;
+
+    let urlSelectedGranules = urlTotalGranules;
+
+    // Gets the total amount of granules that the layer has
+    totalGranules = await fetch(urlTotalGranules, { timeout: 5000 })
+      .then(async(response) => {
+        const result = await response.json();
+        return result.feed.facets.children[0].children[0].children[0].count;
+      })
+      .catch((error) => 0);
+
+    // Gets the total subset of granules that are within the defining bounding box
+    if (showBoundingBox) {
+      urlSelectedGranules += `&bounding_box=${currentExtent.southWest},${currentExtent.northEast}`;
+      selectedGranules = await fetch(urlSelectedGranules, { timeout: 5000 })
+        .then(async(response) => {
+          const result = await response.json();
+          return result.feed.facets.children[0].children[0].children[0].count;
+        })
+        .catch((error) => 0);
+    }
+
+    this.setState({ selectedGranules, totalGranules });
+    this.setState({ isSearchingForGranules: false });
```

Comment:
These two `setState()` calls can be combined into one:
```suggestion
this.setState({
    selectedGranules,
    totalGranules,
    isSearchingForGranules: false,
});
```